src/components/_tests_/Camera.spec.ts

```diff
@@ -0,0 +1,186 @@
+            const camera = fullRenderCamera(defaultProps);
+            const cameraInstance = camera.instance() as any;
+            const takePicture = spyOn(cameraInstance, "takePicture").and.callThrough();
+
+            camera.setProps({ caption: "buttons" , width: 70, height: 50 , widthUnit: "pixels" , heightUnit: "pixels" });
+            camera.setState({ browserSupport: true, pictureTaken: false, screenshot: "" });
+            camera.find(".widget-camera-picture").simulate("click");
+
+            expect(takePicture).toHaveBeenCalledTimes(1);
+        });
+    });
+
+    describe("with multiple media devices", () => {
+        it("should render with 'switch icon'", () => {
+            createCamera.setProps({ filter: "none", caption: "icons" });
+            createCamera.setState({ availableDevices: [ "deviceOne", "deviceTwo" ] });
+
+            expect(createCamera.find({ spanClass: "widget-camera-switch-button" }).prop("glyphIcon")).toBe(defaultProps.switchCameraIcon);
+        });
+
+        it("should render with 'switch button'", () => {
+            createCamera.setProps({ filter: "none", caption: "buttons" });
+            createCamera.setState({ availableDevices: [ "deviceOne", "deviceTwo" ] });
+
+            expect(createCamera.find({ spanClass: "widget-camera-switch-button" }).prop("buttonLabel")).toBe("Switch");
+        });
+
+        it("should switch/swap camera when switch camera button is clicked", () => {
+            const camera = fullRenderCamera(defaultProps);
+            const cameraInstance = camera.instance() as any;
+            const changeCamera = spyOn(cameraInstance, "changeCamera").and.callThrough();
+
+            camera.setProps({ filter: "none", width: 70, height: 50 });
+            camera.setState({ availableDevices: [ "deviceOne", "deviceTwo" ] , browserSupport: true, pictureTaken: false, screenshot: "" });
+            camera.find(".widget-camera-switch-button").simulate("click");
+
+            expect(changeCamera).toHaveBeenCalled();
+        });
+    });
+
+    describe("with icons configuration", () => {
+        it("should render with the correct 'capture icon'", () => {
+            createCamera.setProps({ caption: "icons" });
+            expect(camerabuttons.prop("glyphIcon")).toBe(defaultProps.captureIcon);
+        });
+
+        it("should render with the correct 'retake icon'", () => {
+            createCamera.setProps({ caption: "icons" });
+            createCamera.setState({ pictureTaken: true, screenshot: "base64image string" });
+            expect(camerabuttons.prop("glyphIcon")).toBe(defaultProps.captureIcon);
+        });
+
+        it("should render with the correct 'save icon'", () => {
+            createCamera.setProps({ caption: "icons" });
+            createCamera.setState({ pictureTaken: true, screenshot: "base64image string" });
+            expect(createCamera.find({ spanClass: "widget-camera-switch-button" }).prop("glyphIcon")).toBe(defaultProps.savePictureIcon);
```

Comment:
Test should separated into 2 or 3 parts
-Initialization / Declaration
-Execution
-Tests

So add space

Reply:
Done